### ISSUE TITLE
"Property never defined" warnings on struct with NTI

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#875]
With new type inference I get warnings "Cannot add a property to a struct instance after it is constructed" or "Property never defined" on properties that were not assigned a value in the constructor.
```javascript
/**
@constructor
@struct
*/
MyClass = function() {
  /** @type {number} */
  this.myNumber;
};

var myInstance = new MyClass();
if (!goog.isNumber(myInstance.myNumber))
  myInstance.myNumber = 42;
```
The warnings disappear when I give the property an initial value in the constructor like this:
```javascript
/**
@constructor
@struct
*/
MyClass = function() {
  /** @type {number} */
  this.myNumber = 0;
};
```
Assuming the error is correct, I think the wording of the error message could be improved, because I thought that the property *was* defined.  Perhaps say something about an uninitialized property?  Or ideally detect the error on the line in the constructor where the property is not initialized.

**Comment by myphysicslab** [comment]
I tried to fix these warnings by initializing each property.  That worked for many cases, but not all.  The cases where I'm still getting the warnings are similar to issue #850, in that the assignment is inside an "if" statement, or "for" loop.

**Comment by dimvar** [comment]
I just tried this example and it worked as expected (no warnings). I'm closing this issue, and I'll fix #850 separately.

<code>
/**
 * @constructor
 * @struct
 */
function Foo() {
  /** @type {number} */
  this.prop;
}

(new Foo).prop = 123;
</code>

**Comment by myphysicslab** [comment]
Not seeing these problems any more with Version: v20150505-58-g39bb337
